### ISSUE TITLE
New Work - Model Implementation (III)

### ISSUE THREAD

**Issue by eeshan15** [The-Red-Wood-Lab/Micro-Gas-Turbine-electrical-energy-Prediction#10]
Download Datasets
Train Dataset Link : [Click on this Link](https://github.com/The-Red-Wood-Lab/Micro-Gas-Turbine-electrical-energy-Prediction/blob/main/Dataset/final_merged_file.csv)

Test Dataset Link : [Click on this Link](https://github.com/The-Red-Wood-Lab/Micro-Gas-Turbine-electrical-energy-Prediction/tree/main/Dataset/test) . Please Download both CSV files for test dataset

About The Model Selection
So in our model selection Process here we are with gradient Boosting models. Gradient Boosting models are really great when it comes to predict from the given data because they use learning rate method. Also There are Two types of gradient Boosting Models :

XGBoost : XGBoost builds new models to correct the errors made by previous models, specifically focusing on the gradients (errors) of the loss function. It uses gradient descent to minimize the errors. Each subsequent model (usually a decision tree) corrects the residuals or errors of the previous model.

AdaBoost : AdaBoost focuses on adjusting the weights of the training data points. Initially, it assigns equal weights to all data points. After each iteration, it increases the weights of the data points that were misclassified (or had higher error in regression). This forces the next model to pay more attention to the mistakes of the previous model. The final prediction is a weighted sum of the models’ outputs

In this Issue we are implementing XGBOOST Algorithim.

Implementation of XGBOOST for this issue
Take the training dataset and take 75:25 ratio convert 25% of training into Validation dataset. You can also use K-Fold Cross validation method.

Next Check for overfitting by Visualizing training and Validation loss by plotting them in one single graph Graph. If your Validation Curve is going up and Training curve going down then there is a problem you have to check for the issue. You can use Regularization techniques etc to solve the problem.

Take the learning rate close to 0.1 you can hit and trial as well.

Now here comes the main game , for test dataset introduce a column Voltage_predicted and try predicting values learned by the model so that it can generate predicted values in that column.

Final Evaluation is done by R2 score (R-Squared Score) and RMSE (Root Mean Square Error). If your R2 score is close to 1 and RMSE score is lower. That means your model is good.

Please Provide a difference between Predicted values and actual Values by generating graphs such as Line plot or a scatter plot. The graph should clearly tells plot difference between two values and should be titled as "Predicted vs Actual Value"

## Requirements for Pull Request (PR) Submission.
Your PR should have detailed Description , appropriate title and screen shots of code snippets and graphs
Also please tell the learning rate , R2 score and RMSE score separately in the form of heading in PR.

## Checklists
 

- [ ] Did Visualization
- [ ]  Showcased Code Snippets
- [ ]  Showcased learning Rate , R2 Score and RMSE score
- [ ]  Dealt with Overfitting (if persist)

Note : In your PR you can generate Checklist easily. Provided is the screen shot where the icon lies for automatic check list generation.
Please make sure you submit your PR with this checklists compulsorily.

**Comment by Hotchapu13** [comment]
please assign this to me.

**Comment by eeshan15** [comment]
@Hotchapu13 assigned
